book/src/input/how_to_define_custom_control_bindings.md

```diff
@@ -0,0 +1,172 @@
+# How to Define Custom Control Bindings
+
+Instead of using `StringBindings` for an `InputBundle` you probably want to use a custom type in production, as `StringBindings` are mainly meant to be used for prototyping and not very efficient.
+
+Using a custom type to handle input instead of using `String` has many advantages:
+
+* A `String` uses quite a lot of memory compared to something like an enum.
+* Inputting a `String` when retrieving input data is error-prone if you misstype it or change the name.
+* You will be able to add additional information to a custom type.
+
+## Defining a Custom Type
+
+Defining a custom type for the `InputBundle` is done by implementing the `BindingTypes` trait. This trait contains two types, an `Axis` type and an `Action` type. These types are usually defined as enums, but you can use whatever you want.
+
+```rust,edition2019,no_run,noplaypen
+use serde::{Serialize, Deserialize};
+use amethyst::input::{BindingTypes, Bindings};
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq, Serialize, Deserialize)]
+enum AxisBinding {
+  Horizontal,
+  Vertical,
+}
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq, Serialize, Deserialize)]
+enum ActionBinding {
+  Shoot,
+}
+
+impl Display for AxisBinding {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl Display for ActionBinding {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+struct MovementBindingTypes;
+
+impl BindingTypes for MovementBindingTypes {
+  type Axis = AxisBinding;
+  type Action = ActionBinding;
+}
+```
+
+The `Axis` and `Action` type both need to derive all the traits listed above, the first five are used by Amethyst and the last two are for reading and writing to files correctly. They also need to implement `Display` if you want to add a bindings config file.
+
+For serializing and deserializing you need to add [serde](https://crates.io/crates/serde) to the dependencies like this:
+
+```rust,ignore
+  serde = { version="1.0", features = ["derive"] }
```

Comment:
Is it worth to note here that the latest version might not be `1.0` (it already isn't) by the time someone actually reads this?

Reply:
I don't know if that's necessary, but it wouldn't hurt to change it.

Reply:
I don't know either, honestly. There's a link to the crate above, so it's not like they can't figure out what the latest version is, I just thought it might help newbies to the crate system as a whole to mention that a newer version _might_ be available. Then again, what guarantees that the newer version is compatible with Amethyst, right?